pagopa/io-app
# chore: [IOCOM-1625] FIMS history screen UI revamp

## Short description
updated FIMS History screen to better follow the new design
<img width="250 " alt="Screenshot 2024-09-11 at 17 37 16" src="https://github.com/user-attachments/assets/128d0ee3-632c-474a-9158-915b36baa174"><img width="250" alt="Screenshot 2024-09-11 at 17 37 00" src="https://github.com/user-attachments/assets/a5840b0e-2e06-4d3e-b107-bd9ae5a8f5d6">


## List of changes proposed in this pull request
- moved to `FlashList` to improve performance
- updated components to better fit design
- extrapolated a new `PressableBase` utility component, which was necessary for styling purposes
- strong height lock for the listItems for consistency's sake (also to help `FlashList`'s size precalculations)
- refactors

## How to test
using the dev-server, navigate to the FIMS history and make sure everything correctly follows design -- and also that nothing has broken
